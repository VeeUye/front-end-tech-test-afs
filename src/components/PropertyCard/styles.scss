@import '../../styles/settings.scss';

// ==========================================================================
// Property card
// ==========================================================================

//
// Start adding your styles here!
//


.property-card {
  position: relative;
  display: flex;
  width: 791px;
  height: 210px;
  margin-bottom: 2rem;
  background: #ffffff;
  border: 2px solid #cdccd1;
  border-radius: 8px;
}

.property-card:hover {
  border: 2px solid #ec6b1e;
  border-radius: 8px;
}

// photo
// ===========================================================================

.photo {
  position: absolute;
  width: 300px;
  height: 210px;
  left: 0px;
  top: 0px;
  bottom: 0px;
  border-radius: 8px;
}

// property info
// ===========================================================================

.property-info {
  position: absolute;
  left: 300px;
  right: 154px;
  top: 0px;
  bottom: 0px;
}

.property-info-inner {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 0px;
  position: absolute;
  width: 143px;
  height: 53px;
  left: 24px;
  top: 13px;
}

.frame-106 {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  padding: 0px;
  gap: 6px;
  width: 123px;
  height: 26px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.property-type {
  width: 44px;
  height: 26px;
  font-style: bold;
  font-weight: 500;
  font-size: 14px;
  line-height: 26px;
  display: flex;
  align-items: center;
  color: #0e9685;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.seperator {
  width: 7px;
  height: 26px;
  font-style: normal;
  font-weight: 400;
  font-size: 15px;
  line-height: 26px;
  display: flex;
  align-items: center;
  color: #3c3a4a;
  opacity: 0.32;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.account-type {
  width: 200px;
  height: 26px;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 26px;
  display: flex;
  align-items: center;
  color: #3c3a4a;
  flex: none;
  order: 2;
  flex-grow: 0;
}

.address-header {
  width: 300px;
  height: 27px;
  font-style: normal;
  font-weight: 700;
  font-size: 19.2px;
  line-height: 26px;
  display: flex;
  align-content: flex-start;
  color: #3c3a4a;
  flex: none;
  order: 1;
  flex-grow: 0;
}

// beds and baths
// ===========================================================================

.beds-baths {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0px;
  gap: 24px;
  position: absolute;
  width: 207px;
  height: 24px;
  left: 0px;
  top: 60px;
}

.beds {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  padding: 0px;
  gap: 10px;
  width: 88px;
  height: 24px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.bed-4-1 {
  width: 24px;
  height: 24px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.beds-icon-container {
  position: absolute;
  width: 24px;
  height: 24px;
  left: 0%;
  right: 0%;
  top: 10.35%;
  bottom: 10.35%;
}

.beds-icon {
  position: absolute;
  left: 0%;
  right: 0%;
  top: 3%;
  bottom: 20.35%;
}

.beds-text {
  width: 54px;
  height: 24px;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 19px;
  display: flex;
  align-items: center;
  color: #3c3a4a;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.baths {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  padding: 0px;
  gap: 10px;
  width: 95px;
  height: 24px;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.baths-text {
  width: 61px;
  height: 24px;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 19px;
  display: flex;
  align-items: center;
  color: #3c3a4a;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.shower-icon-container {
  position: absolute;
  width: 20px;
  height: 20px;
  left: 42%;
  right: 0%;
  bottom: 10.35%;
}

.shower-icon {
  position: absolute;
  left: 0%;
  right: 0%;
  top: 10.35%;
  bottom: 10.35%;
}

// info
// ===========================================================================

.info {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 0px;
  gap: 12px;
  position: absolute;
  width: 206px;
  height: 75px;
  left: 24px;
  top: 119px;
}

.tenancy-agreement {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 8px;
  width: 153px;
  height: 17px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.profile {
  width: 16px;
  height: 16px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.tenancy-agreement-text {
  width: 129px;
  height: 17px;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  display: flex;
  align-items: center;
  color: #777584;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.available-from {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 8px;
  width: 206px;
  height: 17px;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.available-from-text {
  width: 182px;
  height: 17px;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  display: flex;
  align-items: center;
  color: #777584;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.calender {
  width: 16px;
  height: 16px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.address {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 8px;
  width: 198px;
  height: 17px;
  flex: none;
  order: 2;
  flex-grow: 0;
}

.location {
  width: 16px;
  height: 16px;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.location-text {
  width: 174px;
  height: 17px;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  display: flex;
  align-items: center;
  color: #3c3a4a;
  flex: none;
  order: 1;
  flex-grow: 0;
}

// heart
// ===========================================================================

.heart {
  position: absolute;
  width: 24px;
  height: 24px;
  right: 16px;
  top: 16px;
}

.heart-icon {
  position: absolute;
  left: 4.17%;
  right: 4.17%;
  top: 8.33%;
  bottom: 8.33%;
}

// price
// ===========================================================================

.price {
  position: absolute;
  width: 171px;
  height: 71px;
  right: 16px;
  bottom: 16px;
}

.price-inner {
  flex-direction: column;
  align-items: flex-start;
  padding: 0px;
  position: absolute;
  width: 107px;
  height: 37px;
  right: 0px;
  bottom: 30px;
}

.price-text {
  width: 36px;
  height: 13px;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 13px;
  display: flex;
  align-items: center;
  letter-spacing: 0.01em;
  color: #3c3a4a;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.price-value {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: flex-end;
  padding: 0px;
  gap: 2px;
  width: 107px;
  height: 31px;
  flex: none;
  order: 1;
  flex-grow: 0;
}

.price-ppw {
  width: 57px;
  height: 31px;
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 30px;
  display: flex;
  align-items: center;
  color: #ec6b1e;
  flex: none;
  order: 0;
  flex-grow: 0;
}

.bills {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 4px;
  position: absolute;
  width: 131px;
  height: 17px;
  right: -0.5px;
  top: 54px;
}

.bills-inner {
  width: 131px;
  height: 17px;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  display: flex;
  align-items: center;
  text-align: right;
  color: #636171;

  flex: none;
  order: 1;
  flex-grow: 0;
}
