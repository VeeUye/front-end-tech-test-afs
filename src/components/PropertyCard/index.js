import React from 'react'
import PropTypes from 'prop-types'

import BedIcon from '../icons/BedIcon'
import FavouriteIcon from '../icons/FavouriteIcon'
import BathIcon from '../icons/BathIcon'
import AvailableFromIcon from '../icons/AvailableFromIcon'
import DistanceFromIcon from '../icons/DistanceFromIcon'
import TenancyAgreementIcon from '../icons/TenancyAgreementIcon'
import '../PropertyCard/styles.scss'

const PropertyCard = ({ properties }) => {
  const {
    accountType,
    address,
    availableFrom,
    baths,
    beds,
    distanceFromUniversity,
    propertyType,
    tenancyAgreement,
    terms,
    thumbnails,
  } = properties[0]

  return (
    // <div className="property-card">
    //   <img className="photo" src={thumbnails[0].url} alt="property"></img>
    //   <div className="property-info">
    //     <div className="propert-info-innner">
    //       <div className="frame-106">
    //         <div className="property-type">{propertyType}</div>
    //       </div>
    //     </div>
    //   </div>
    // </div>

    <div className="property-card">
      <img className="photo" src={thumbnails[0].url} alt="property"></img>
      <div className="property-info">
        <div className="property-info-inner">
          <div className="frame-106">
            <div className="property-type">{propertyType}</div>
            <div className="seperator"> • </div>
            <div className="account-type">{accountType}</div>
          </div>
          <div className="address-header">{address.address2}</div>
          <div className="beds-baths">
            <div className="beds">
              <div className="bed-4-1">
                <div className="beds-icon-container">
                  <div className="beds-icon">
                    <BedIcon />
                  </div>
                </div>
              </div>
              <div className="beds-text">{beds} bed</div>
            </div>
            <div className="baths">
              <div className="shower-icon-container">
                <div className="shower-icon">
                  <BathIcon />
                </div>
              </div>

              <div className="baths-text">{baths} bath</div>
            </div>
          </div>
        </div>
        <div className="info">
          <div className="tenancy-agreement">
            <div className="profile">
              <TenancyAgreementIcon />
            </div>
            <div className="tenancy-agreement-text">
              Suitable for {tenancyAgreement}
            </div>
          </div>
          <div className="address">
            <div className="location">
              <DistanceFromIcon />
            </div>
            <div className="location-text">
              {distanceFromUniversity.time}&nbsp;
              {distanceFromUniversity.modeOfTransport} to University
            </div>
          </div>
          <div className="available-from">
            <div className="calender">
              <AvailableFromIcon />
            </div>
            <div className="available-from-text">
              Available from {availableFrom}
            </div>
          </div>
        </div>
      </div>

      <div className="heart">
        <FavouriteIcon />
      </div>
      <div className="price">
        <div className="price-inner">
          <div className="price-text">FROM</div>
          <div className="price-value">
            <div className="price-ppw">£{terms.rentPpw.value}</div>&nbsp;pp/pw
          </div>
        </div>
        <div className="bills">
          <div className="bills-inner">
            {terms.billsIncluded}&nbsp;bills included
          </div>
        </div>
      </div>
    </div>
  )
}

export default PropertyCard

PropertyCard.propTypes = {
  props: PropTypes.arrayOf(
    PropTypes.shape({
      accountType: PropTypes.string,
      address: PropTypes.shape({
        address2: PropTypes.string,
      }),
      availableFrom: PropTypes.string,
      baths: PropTypes.number,
      beds: PropTypes.number,
      distanceFromUniversity: PropTypes.shape({
        time: PropTypes.string,
        modeOfTransport: PropTypes.string,
      }),
      propertyType: PropTypes.string,
      tenancyAgreement: PropTypes.string,
      terms: PropTypes.shape({
        bills: PropTypes.string,
        rentPpw: PropTypes.shape({
          value: PropTypes.number,
          currency: PropTypes.string,
        }),
      }),
      thumbnails: PropTypes.arrayOf(
        PropTypes.shape({
          url: PropTypes.string,
        })
      ),
    })
  ),
}
