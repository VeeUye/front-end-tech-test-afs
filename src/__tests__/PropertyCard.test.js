import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import PropertyCard from '../components/PropertyCard'

describe('PropertyCard', () => {
  const validProps = {
    accountType: 'Landlord',
    address: {
      address2: '12 Darley Road',
    },
    availableFrom: '12 December',
    baths: 1,
    beds: 2,
    distanceFromUniversity: {
      time: '10 mins',
      modeOfTransport: 'walk',
    },
    propertyType: 'flat',
    tenancyAgreement: 'groups',
    terms: {
      bills: 'some',
      rentPpw: {
        value: 100,
        currency: 'GBP',
      },
    },
    thumbnails: {
      url: 'pic/is/here',
    },
  }

  it('renders correctly', () => {
    const { asFragment } = render(
      <PropertyCard
        accountType={validProps.accountType}
        address={validProps.address}
        availableFrom={validProps.availableFrom}
        baths={validProps.baths}
        beds={validProps.beds}
        distanceFromUniversity={validProps.distanceFromUniversity}
        propertyType={validProps.propertyType}
        tenancyAgreement={validProps.tenancyAgreement}
        terms={validProps.terms}
        thumbnails={validProps.thumbnails.url}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correct values for props', () => {
    const { getByText } = render(
      <PropertyCard
        accountType={validProps.accountType}
        address={validProps.address}
        availableFrom={validProps.availableFrom}
        baths={validProps.baths}
        beds={validProps.beds}
        distanceFromUniversity={validProps.distanceFromUniversity}
        propertyType={validProps.propertyType}
        tenancyAgreement={validProps.tenancyAgreement}
        terms={validProps.terms}
        thumbnails={validProps.thumbnails.url}
      />
    )

    expect(screen.getByText(/Landlord/i)).toBeInTheDocument()

    expect(screen.getByText(/Darley Road/i)).toBeInTheDocument()

    expect(screen.getByText(/12 December/i)).toBeInTheDocument()

    expect(screen.getByText(/walk/i)).toBeInTheDocument()

    expect(screen.getByText(/flat/i)).toBeInTheDocument()

    expect(screen.getByText(/suitable for groups/i)).toBeInTheDocument()

    expect(screen.getByText(/bills included/i)).toBeInTheDocument()
  })
})
