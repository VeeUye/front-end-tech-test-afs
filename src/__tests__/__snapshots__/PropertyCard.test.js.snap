// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <img
      alt="property"
      class="photo"
    />
    <div
      class="property-info"
    >
      <div
        class="property-info-inner"
      >
        <div
          class="frame-106"
        >
          <div
            class="property-type"
          >
            flat
          </div>
          <div
            class="seperator"
          >
             • 
          </div>
          <div
            class="account-type"
          >
            Landlord
          </div>
        </div>
        <div
          class="address-header"
        >
          12 Darley Road
        </div>
        <div
          class="beds-baths"
        >
          <div
            class="beds"
          >
            <div
              class="bed-4-1"
            >
              <div
                class="beds-icon-container"
              >
                <div
                  class="beds-icon"
                >
                  <svg>
                    bed-icon.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="beds-text"
            >
              2 bed
            </div>
          </div>
          <div
            class="baths"
          >
            <div
              class="shower-icon-container"
            >
              <div
                class="shower-icon"
              >
                <svg>
                  bath-icon.svg
                </svg>
              </div>
            </div>
            <div
              class="baths-text"
            >
              1 bath
            </div>
          </div>
        </div>
      </div>
      <div
        class="info"
      >
        <div
          class="tenancy-agreement"
        >
          <div
            class="profile"
          >
            <svg>
              tenancy-agreement-icon.svg
            </svg>
          </div>
          <div
            class="tenancy-agreement-text"
          >
            Suitable for groups
          </div>
        </div>
        <div
          class="address"
        >
          <div
            class="location"
          >
            <svg>
              distance-from-icon.svg
            </svg>
          </div>
          <div
            class="location-text"
          >
            10 mins walk to University
          </div>
        </div>
        <div
          class="available-from"
        >
          <div
            class="calender"
          >
            <svg>
              available-from-icon.svg
            </svg>
          </div>
          <div
            class="available-from-text"
          >
            Available from 12 December
          </div>
        </div>
      </div>
    </div>
    <div
      class="heart"
    >
      <svg>
        favourite-icon.svg
      </svg>
    </div>
    <div
      class="price"
    >
      <div
        class="price-inner"
      >
        <div
          class="price-text"
        >
          FROM
        </div>
        <div
          class="price-value"
        >
          <div
            class="price-ppw"
          >
            £100
          </div>
           pp/pw
        </div>
      </div>
      <div
        class="bills"
      >
        <div
          class="bills-inner"
        >
           bills included
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
